src/core/lv_obj.c

```diff
@@ -309,10 +313,55 @@ bool lv_obj_is_valid(const lv_obj_t * obj)
     return false;
 }
 
+#if LV_USE_OBJ_ID
+void lv_obj_set_id(lv_obj_t * obj, void * id)
+{
+    obj->id = id;
+}
+
+void * lv_obj_get_id(const lv_obj_t * obj)
+{
+    return obj->id;
+}
+
+lv_obj_t * lv_obj_get_child_by_id(const lv_obj_t * obj, void * id, bool (compare)(lv_obj_t *, void * id))
```

Comment:
IMO we shouldn't pass `compare` in each call as it's the same in the whole app. Instead we can expect the user to either 
1. implement it in the app with a  fixed name (e.g. `lv_obj_id_compare`)
2. set it as a global callback
 
As an alternative to use can create a wrapper like:
```
lv_obj_t * my_lv_obj_get_child_by_id(const lv_obj_t * obj, void * id)
{
   return lv_obj_get_child_by_id(obj, id, my_compare);
}
```